src/main/daml/Daml/Finance/Account/Account.daml

```diff
@@ -23,38 +24,62 @@ template Account
     owner : Party
       -- ^ The account owner.
     controllers : Account.Controllers
-      -- ^ Controllers of transfers.
+      -- ^ Controllers of transfers and freezes.
     id : Id
       -- ^ Identifier of the account.
     description : Text
       -- ^ Description of the account.
     holdingFactoryCid : ContractId HoldingFactory.F
       -- ^ Associated holding factory.
+    frozen : Optional Account.Frozen
+      -- ^ Frozen account (if any).
     observers : PartiesMap
       -- ^ Observers.
   where
-    signatory custodian, owner
-    observer Disclosure.flattenObservers observers
+    let
+      account = AccountKey with custodian; owner; id
+      mustNotBeFrozen = assertMsg @Update "account must not be frozen" $ isNone frozen
```

Comment:
Is the type annotation `@Update` needed?

Reply:
Yes, otherwise I see
```
Ambiguous type variable ‘m2’ arising from a use of ‘_templateLet_Account’ prevents the constraint ‘(CanAssert m2)’ from being solved.
```